src/coreclr/vm/cgensys.h

```diff
@@ -115,12 +147,26 @@ inline bool TargetHasAVXSupport()
 #if (defined(TARGET_X86) || defined(TARGET_AMD64))
     int cpuInfo[4];
     __cpuid(cpuInfo, 0x00000001);           // All x86/AMD64 targets support cpuid.
-    return ((cpuInfo[CPUID_ECX] & (1 << 28)) != 0); // The AVX feature is ECX bit 28.
+
+    const int EXPECTED_EDX = (1 << 25)  // SSE
+                           | (1 << 26); // SSE2
+
+    const int EXPECTED_ECX = (1 << 0)   // SSE3
+                           | (1 << 9)   // SSSE3
+                           | (1 << 19)  // SSE4.1
+                           | (1 << 20)  // SSE4.2
+                           | (1 << 27)  // OSXSAVE
+                           | (1 << 28); // AVX
+
+
+    if (((cpuInfo[CPUID_EDX] & EXPECTED_EDX) == EXPECTED_EDX) && ((cpuInfo[CPUID_ECX] & EXPECTED_ECX) == EXPECTED_ECX))
```

Comment:
Instead of duplicating the logic here, would it be better to just call into JIT manager to get the `InstructionSet_AVX` bit from `CPUCompileFlags`?

This logic will get even more complex once we fix https://github.com/dotnet/runtime/issues/60035

Reply:
Probably, yes.

I assumed there was some existing reason for this separation. If there isn't, it's likely more trivial to just reuse what we already have computed (assuming its correctly taking into account things like `crossgen` flags, etc)

Reply:
Actually, we do call into the code manager right after calling `TargetHasAVXSupport` anyway that should take care of everything. I think that the `TargetHasAVXSupport ` is unnecessary and can be just deleted.

It does not explain where the bug may be.

Reply:
> I think that the TargetHasAVXSupport is unnecessary and can be just deleted.

I think you may be right...

> It does not explain where the bug may be.

AFAIK, all of the logic for tracking and checking per-ISA checks (even for crossgen) is supposed to be driven around the information set by `void EEJitManager::SetCpuInfo()` and so if `Avx` isn't set there; the comparison of "is the R2R code supported" should fail if the method is annotated as `uses Avx`.

I'll try and dig in a bit more, but @davidwrighton might be able to give direct pointers to the right checks to look at/debug through.